### PR TITLE
[FX-4620] Migrate Switch component

### PR DESCRIPTION
**Pull request by dmaklygin**
[FX-4620]

### Description

Migrate Switch component to Mui/base Switch and TailwindCSS.

### How to test

- [Temploy](https://picasso.toptal.net/fx-4620-migrate-switch)

### Video recording

This is a comparison between Temploy and production:

https://github.com/toptal/picasso/assets/1824723/f2496ff5-a060-4d10-8afd-39d234fe50ac

### Accessibility issues.

No new accessibility issues. Current issues are related to disabled uncontrolled Switch. The same as on production.

Screenshot 2024-05-14 at 10 41 54

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- N/A codemod is created and showcased in the changeset
- N/A test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-4620]: https://toptal-core.atlassian.net/browse/FX-4620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/base/Switch/src/Switch/Switch.tsx [hunk 1]
```diff
@@ -60,9 +58,38 @@ export const Switch = forwardRef<HTMLButtonElement, Props>(function Switch(
       style={style}
       disabled={disabled}
       id={id}
-      onChange={onChange}
-      value={value}
```
Comment (dmaklygin) [inline packages/base/Switch/src/Switch/Switch.tsx:64]: value is completely useless for Switch

File: packages/picasso-forms/src/Switch/Switch.tsx [hunk 2]
```diff
@@ -34,8 +36,14 @@ export const Switch = (props: Props) => (
       // omitting highlight from inputProps
       // eslint-disable-next-line @typescript-eslint/no-unused-vars
       highlight,
+      // value is not expected in PicassoSwitch
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      value,
```
Comment (dmaklygin) [inline packages/picasso-forms/src/Switch/Switch.tsx:41]: exclude `value` from passing to `Switch`

**Comment by dmaklygin** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]
Looks good, I think we should also increase peerDependency of `@toptal/picasso-tailwind` on `@toptal/picasso` package.json as if somebody will be using `import { Switch } from '@toptal/picasso'` it will not work without new version

File: .changeset/few-cobras-agree.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': minor
```
Comment (TomasSlama) [inline .changeset/few-cobras-agree.md:2]:
```suggestion
'@toptal/picasso': major
```

this is not new feature, users need to take a action and update the dependency of picasso-tailwind

Comment (dmaklygin) [reply .changeset/few-cobras-agree.md:2]: The current picasso version utilizes "@toptal/picasso-button": "2.0.1", which already needs "@toptal/picasso-tailwind": "^2.5.0"
For me,  it means, the Picasso has already expect to have `picasso-tailwind` integrated in the project.
WDYT?

Comment (TomasSlama) [reply .changeset/few-cobras-agree.md:2]: it means we didn't release major release for the Picasso, right?

File: packages/base/Switch/package.json [hunk 4]
```diff
@@ -32,6 +34,7 @@
   "peerDependencies": {
     "@material-ui/core": "4.12.4",
```
Comment (TomasSlama) [inline packages/base/Switch/package.json:35]: can we get rid of this

File: .changeset/few-cobras-agree.md (hunk 3 above)
Comment (dmaklygin) [reply .changeset/few-cobras-agree.md]: Updated to `major` release as we decided! :)

File: packages/base/Switch/src/Switch/Switch.tsx [hunk 5]
```diff
@@ -60,9 +58,38 @@ export const Switch = forwardRef<HTMLButtonElement, Props>(function Switch(
...
+      onChange={onChangeCallback}
```
Comment (OleksandrNechai) [inline packages/base/Switch/src/Switch/Switch.tsx:61]: Why this change is needed?

File: packages/picasso-tailwind/src/index.js [hunk 6]
```diff
@@ -118,6 +118,7 @@ module.exports = {
       white: '#FFFFFF',
       black: '#000000',
       transparent: 'transparent',
+      current: 'currentColor',
```
Comment (OleksandrNechai) [inline packages/picasso-tailwind/src/index.js:121]: Could you please explain this change?

**Review by OleksandrNechai — APPROVED** [review]

**Review by OleksandrNechai — APPROVED** [review]
IMO good to go. I have only questions, ping me please when answered. Thanks! :+1:

**Comment by dmaklygin** [comment]
The previous version has a bug due to hidden input: 
Screenshot 2024-05-14 at 17 35 11

This was a reason why happo tests were fallen

File: packages/base/Switch/src/Switch/Switch.tsx (hunk 5 above)
Comment (dmaklygin) [reply packages/base/Switch/src/Switch/Switch.tsx:61]: As I said, MUI/Base Switch doesn't expect value, and expecting value in the Picasso's Swich is also odd.
It should keep only state, nothing more.
`value` was added unintentionally to support consistency in Field elements, so, better to remove it.

File: packages/picasso-tailwind/src/index.js (hunk 6 above)
Comment (dmaklygin) [reply packages/picasso-tailwind/src/index.js:121]: this approach was migrated from the previous implementation:
we applied it to Switch: `text-white bg-current`: it means, we say that background-color will be the same as the text color -> white.

The default TailwindCSS configuration contains such property:

https://github.com/tailwindlabs/tailwindcss/blob/next/packages/tailwindcss/src/utils/is-color.ts#L174

Comment (OleksandrNechai) [reply packages/picasso-tailwind/src/index.js:121]: yeah, that is what was confusing to me. Default config of tailwind has this bg-current. Why did we have to add our own?

File: packages/base/Switch/src/Switch/Switch.tsx (hunk 5 above)
Comment (OleksandrNechai) [reply packages/base/Switch/src/Switch/Switch.tsx:61]: I am asking about `onChange={onChange}` why we had to change it
